### PR TITLE
[HttpClient] Avoid installation of latest contracts as it's not compatible

### LINKED ISSUE #40975: SF4.4 - should implement method HttpClientInterface::withOptions(array $options)
**Symfony version(s) affected**: 4.4.21

**Description**
when executing my Functional tests:

1x: Class "Symfony\Component\HttpClient\TraceableHttpClient" should implement method "Symfony\Contracts\HttpClient\HttpClientInterface::withOptions(array $options)": Returns a new instance of the client with new default options

**How to reproduce**
instance any http client in sf 4.4.21

**Possible Solution**
back port the fix that has be done in 5.3

**Additional context**
https://github.com/symfony/symfony/issues/40893

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | ref #40975 #40996 #40893 
| License       | MIT
| Doc PR        |

To be reverted on 5.x (5.3)

Fixes #40975

### CODE REVIEW

**Comment by OskarStark** [comment]
I added a Fixes... for the open ticket to the PR header

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/HttpClient/composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "require": {
         "php": ">=7.1.3",
         "psr/log": "^1.0",
-        "symfony/http-client-contracts": "^1.1.10|^2",
+        "symfony/http-client-contracts": "^1.1.10|<2.4",
```
Comment (stof) [inline src/Symfony/Component/HttpClient/composer.json:26]: should be `^1.1.10|>=2,<2.4`. Otherwise, the `<2.4` constraint allows to install `1.0.0`

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
HttpClient *is* compatible with Contracts 2.4. It "just" triggers a deprecation when used with it, but that can be ignored.

I think we should try patching `DebubClassLoader` instead.

**Comment by lyrixx** [comment]
It just trigger deprecation notice because we ensure to not break BC. That's why we added `@method` annotation, and not the method directly. So indeed it's compatibly because ut "just" trigger deprecation. And there is no way to avoid them on 4.4

**Comment by nicolas-grekas** [comment]
Closing in favor of #41075
Thanks for the patch!

**Pull request closed without merging**